File: javascript/5_kyu/simple_pig_latin.js
// Simple Pig Latin
// https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/javascript

// Solution 1: Index and Append
function isAlpha(word) {
    return /^[a-zA-Z]+$/.test(word);
}

function pigIt(str) {
    str = str.split(" ");
    var ans = [];
    
    for (var i = 0; i < str.length; i++) {
        if ( isAlpha(str[i]) ) {
            var switched = str[i].slice(1).concat(str[i].slice(0, 1));
            ans.push(switched.concat("ay"));
        } else {
            ans.push(str[i]);
        }
    }

    return ans.join(" ");
}

console.log(pigIt("Pig latin is cool"));
console.log(pigIt("This is my string"));
console.log(pigIt("Hello world !"));

File: python/5_kyu/simple_pig_latin.py
# Simple Pig Latin
# https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/python

# Solution 1: Split and Append
def split(word):
    return [i for i in word]

def pig_it(s):
    s = s.split(" ")
    ans = []
    
    for i in s:
        if i.isalpha() == True:
            char_list = split(i)
            move = char_list.pop(0)
            char_list.append(move)
            ans.append("".join(char_list) + "ay")
        else:
            ans.append(i)
                 
    return " ".join(ans)

print(pig_it("Pig latin is cool"))
print(pig_it("Hello World!"))
print(pig_it("This is my string"))